domain/src/test/java/org/oppia/domain/profile/ProfileManagementControllerTest.kt

```diff
@@ -95,8 +122,392 @@ class ProfileManagementControllerTest {
 
   @Test
   @ExperimentalCoroutinesApi
-  fun testProfileManagementController() = runBlockingTest(coroutineContext) {
-    // TODO(#16): Finish test cases with full implementation
+  fun testAddProfile_addProfile_checkProfileIsAdded() = runBlockingTest(coroutineContext) {
+    profileManagementController.addProfile("James", "123", null, true).observeForever(mockUpdateResultObserver)
+    advanceUntilIdle()
+
+    val profileDatabase = ProfileDatabase.parseFrom(FileInputStream(File(context.filesDir, "profile_database.cache")))
```

Comment:
Here & elsewhere: we should probably use ``use`` to make sure the stream is closed. Maybe use a helper:

```kotlin
private fun readProfileDatabase(): ProfileDatabase {
  return FileInputStream(File(context.filesDir, "profile_database.cache")).use(ProfileDatabase::parseFrom)
}
```

Reply:
Updated to use the suggested function.